### PR TITLE
Testing Rework

### PR DESCRIPTION
**Pull request by naterush**
- Adds + moves fixtures around
- Tests all protocols that we can test
- Standardized how we create tests for a protocol (for organization!)
- Run safety oracle tests against all safety oracles for all protocols

TODO:
- Add ability to specify initial bets, so we can really test some things we can't currently
.

### CODE REVIEW

**Comment by naterush** [comment]
A brief, initial review would be greatly appreciated. Have some work left to do, see TODO above.

**Comment by naterush** [comment]
Going to leave the initial messages stuff to the next PR. The issue is still open. I'd like to get this merged and then cut another release.

File: tests/casper/protocols/blockchain/test_blockchain_message.py [hunk 1]
```diff
@@ -9,6 +9,37 @@
...
+    [
+        (None, True),
+        ('block', False),
+        (0, False),
+        (True, False),
+    ]
+)
+def test_accepts_valid_estimates(estimate, is_valid, block):
+    if estimate == 'block':
+        Block.is_valid_estimate(block) == is_valid
```
Comment (djrtwo) [inline tests/casper/protocols/blockchain/test_blockchain_message.py]: probably `return` after this check so you aren't inadvertently testing the estimate -- `"block"` with `is_valid` -- `False`. Even though it happens to agree, it's probably not what you meant to test.

If you want to test that a string fails, I'd do a separate test case against `"some random string taht isn't block"`

File: casper/protocols/concurrent/block.py [hunk 2]
```diff
@@ -5,12 +5,11 @@
...
 
-    def __init__(self, estimate, justification, sender, sequence_number, display_height):
-        # Do some type checking for safety!
-        assert isinstance(estimate, dict), "...expected a valid estimate!"
-        assert len(estimate['blocks']) > 0, "... most point to a least one thing"
-
-        super().__init__(estimate, justification, sender, sequence_number, display_height)
+    @classmethod
+    def is_valid_estimate(cls, estimate):
+        if not isinstance(estimate, dict) and len(estimate['blocks']) > 0:
```
[the remark this answers is not in the document]
Comment (djrtwo) [reply casper/protocols/concurrent/block.py]: There is also no file test_concurrent_block.py where this should be tested!

**Comment by naterush** [comment]
Comments addressed. The is_valid_estimate method for concurrent got a little big - but I can't figure out any way to check all the necessary conditions otherwise. 

Pending review, LGTM!

File: casper/protocols/concurrent/block.py [hunk 3]
```diff
@@ -5,12 +5,26 @@
...
 
-    def __init__(self, estimate, justification, sender, sequence_number, display_height):
-        # Do some type checking for safety!
-        assert isinstance(estimate, dict), "...expected a valid estimate!"
-        assert len(estimate['blocks']) > 0, "... most point to a least one thing"
-
-        super().__init__(estimate, justification, sender, sequence_number, display_height)
+    @classmethod
+    def is_valid_estimate(cls, estimate):
+        if not isinstance(estimate, dict):
```
Comment (djrtwo) [inline casper/protocols/concurrent/block.py]: Remove print statements!

**Pull request merged**